src/encoded/tests/test_search.py

```diff
@@ -1,12 +1,170 @@
 # Use workbook fixture from BDD tests (including elasticsearch)
-from .features.conftest import app_settings, app, workbook
+from collections import OrderedDict
 from webob.multidict import MultiDict
 import pytest
+from encoded.tests.features.conftest import app_settings, app, workbook
+from snovault import TYPES
+from snovault.helpers.helper import (
+    set_filters,
+    set_facets,
+    format_facets,
+    sort_query,
+    get_pagination,
+    prepare_search_term,
+    set_sort_order,
+    get_search_fields,
+    list_visible_columns_for_schemas,
+    list_result_fields,
+    build_terms_filter,
+    build_aggregation,
+    normalize_query
+)
 
 
-# Integration tests
+class FakeRequest(object):
+    path = '/search/'
+
+    def __init__(self, params):
+        self.params = MultiDict(params)
+        self.permissions = {}
+
+    def set_permission(self, perm_key, perm_value=False):
+        self.permissions[perm_key] = perm_value
+
+    def has_permission(self, perm_key):
+        if perm_key in self.permissions and self.permissions[perm_key] is True:
+            return True
+        return False
+
+
+@pytest.fixture
+def sort_query_item():
+    item = [({'d': 'val', 'c': 'val',
+              'b': 'val', 'a': 'val'},
+             OrderedDict([('a', 'val'), ('b', 'val'),
+                          ('c', 'val'), ('d', 'val')])),
+            ({'d': {'b': 'val', 'a': 'val'}},
+             OrderedDict([('d', OrderedDict([('a', 'val'),
+                                             ('b', 'val')]))]))]
+    return item
+
+
+@pytest.fixture
+def get_pagination_item():
+    requests = {FakeRequest((('from', '0'), ('limit', '25'))): ('0', 25),
+                FakeRequest((('from', '10'), ('limit', '50'))): ('10', 50),
+                FakeRequest((('from', '0'),)): ('0', 25),
```

Comment:
For whatever reason, if I have just one pair of values, I need to have these in-trailing commas. Otherwise, it throws an error like this:

self = <[ValueError("too many values to unpack (expected 2)") raised in repr()] MultiDict object at 0x10b2070f0>, key = 'from'

    def __getitem__(self, key):
>       for k, v in reversed(self._items):
E       ValueError: need more than 1 value to unpack

eggs/WebOb-1.6.0-py3.4.egg/webob/multidict.py:96: ValueError

It seems like its a property of MultiDict. Not sure.